static/js/components/ObservationPlanRequestLists.jsx

```diff
@@ -0,0 +1,304 @@
+import Accordion from "@material-ui/core/Accordion";
+import AccordionSummary from "@material-ui/core/AccordionSummary";
+import AccordionDetails from "@material-ui/core/AccordionDetails";
+import Typography from "@material-ui/core/Typography";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import {
+  makeStyles,
+  createTheme,
+  MuiThemeProvider,
+  useTheme,
+} from "@material-ui/core/styles";
+import MUIDataTable from "mui-datatables";
+
+import * as Actions from "../ducks/gcnEvent";
+
+const useStyles = makeStyles(() => ({
+  observationplanRequestTable: {
+    borderSpacing: "0.7em",
+  },
+  actionButtons: {
+    display: "flex",
+    flexFlow: "row wrap",
+  },
+  accordion: {
+    width: "99%",
+  },
+  container: {
+    margin: "1rem 0",
+  },
+}));
+
+// Tweak responsive styling
+const getMuiTheme = (theme) =>
+  createTheme({
+    palette: theme.palette,
+    overrides: {
+      MUIDataTable: {
+        paper: {
+          width: "100%",
+        },
+      },
+      MUIDataTableBodyCell: {
+        stackedCommon: {
+          overflow: "hidden",
+          "&:last-child": {
+            paddingLeft: "0.25rem",
+          },
+        },
+      },
+      MUIDataTablePagination: {
+        toolbar: {
+          flexFlow: "row wrap",
+          justifyContent: "flex-end",
+          padding: "0.5rem 1rem 0",
+          [theme.breakpoints.up("sm")]: {
+            // Cancel out small screen styling and replace
+            padding: "0px",
+            paddingRight: "2px",
+            flexFlow: "row nowrap",
+          },
+        },
+        tableCellContainer: {
+          padding: "1rem",
+        },
+        selectRoot: {
+          marginRight: "0.5rem",
+          [theme.breakpoints.up("sm")]: {
+            marginLeft: "0",
+            marginRight: "2rem",
+          },
+        },
+      },
+    },
+  });
+
+const ObservationPlanRequestLists = ({ observationplanRequests }) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const theme = useTheme();
+
+  const [isDeleting, setIsDeleting] = useState(null);
+  const handleDelete = async (id) => {
+    setIsDeleting(id);
+    await dispatch(Actions.deleteObservationPlanRequest(id));
+    setIsDeleting(null);
+  };
+
+  const { instrumentList, instrumentFormParams } = useSelector(
+    (state) => state.instruments
+  );
+
+  if (
+    !instrumentList ||
+    instrumentList.length === 0 ||
+    Object.keys(instrumentFormParams).length === 0
+  ) {
+    return <CircularProgress />;
+  }
+
+  if (!observationplanRequests || observationplanRequests.length === 0) {
+    return <p>No observation plan requests for this source...</p>;
+  }
+
+  const instLookUp = instrumentList.reduce((r, a) => {
+    r[a.id] = a;
+    return r;
+  }, {});
+
+  const requestsGroupedByInstId = observationplanRequests.reduce((r, a) => {
+    r[a.allocation.instrument.id] = [
+      ...(r[a.allocation.instrument.id] || []),
+      a,
+    ];
+    return r;
+  }, {});
```

Comment:
Some Javascript mastery over here!

Reply:
hahaha I expect this was Danny setting up the FollowupList!